src/nxt_sockaddr.c

```diff
@@ -588,9 +590,14 @@ nxt_sockaddr_unix_parse(nxt_mp_t *mp, nxt_str_t *addr)
     const size_t max_len = sizeof(struct sockaddr_un)
                            - offsetof(struct sockaddr_un, sun_path) - 1;
 
+    str = nxt_str_dup(mp, NULL, addr);
+    if (nxt_slow_path(str == NULL)) {
+        return NULL;
+    }
+
     /* Cutting "unix:". */
-    length = addr->length - 5;
-    path = addr->start + 5;
+    length = str->length - 5;
+    path = str->start + 5;
```

Comment:
Heh, I think I must be being dense!

But why can't we just use addr directly?

Reply:
```c
$ grepc -tfd nxt_conf_vldt_server .
./src/nxt_conf_validation.c:static nxt_int_t
nxt_conf_vldt_server(nxt_conf_validation_t *vldt, const nxt_rstr_t *name,
    nxt_conf_value_t *value)
{
    ...

    sa = nxt_sockaddr_parse(vldt->pool, name);

    ...
}
```

`name` is a config string, read-only in my tree.  Showing where it comes from would be complex, but let's say it is mapped to the config (if you edit it, you'll see).

Then `nxt_sockaddr_parse()` will pass that `name` to what `nxt_sockaddr_unix_parse()` knows as `addr`.

Reply:
I.e why do we need to strdup it?

Reply:
Try not to dup it, and inspect `$prefix/var/lib/unit/conf.json` :)

Reply:
```
        $ sudo strace -f -e 'bind,connect' /opt/local/nginx/unit/d8e0/sbin/unitd \
        |& grep abstract;
        [pid 22406] bind(10, {sa_family=AF_UNIX, sun_path=@"abstract\0"}, 12) = 0
        [pid 22410] connect(134, {sa_family=AF_UNIX, sun_path=@"abstract\0"}, 12) = 0
        ^C
        $ sudo killall unitd
        $ sudo strace -f -e 'bind,connect' /opt/local/nginx/unit/master/sbin/unitd \
        |& grep abstract;
        [pid 22449] bind(10, {sa_family=AF_UNIX, sun_path=@"abstract"}, 11) = 0
        [pid 22453] connect(134, {sa_family=AF_UNIX, sun_path=@"abstract\0"}, 12) = -1 ECONNREFUSED (Connection refused)
        ^C
        $ sudo killall unitd
        $ sudo strace -f -e 'bind,connect' /opt/local/nginx/unit/unix/sbin/unitd \
        |& grep abstract;
        [pid 22488] bind(10, {sa_family=AF_UNIX, sun_path=@"abstract"}, 11) = 0
        [pid 22492] connect(134, {sa_family=AF_UNIX, sun_path=@"abstract"}, 11) = 0
        ^C
```

Reply:
See how `master` differs from `unix` (the branch where I've fixed that).

Reply:
And the JSON:

```
        $ sudo cat /opt/local/nginx/unit/master/var/lib/unit/conf.json \
        | jq . \
        | grep unix;
            "unix:@abstract": {
                "unix:\u0000abstract": {}
 
        $ sudo cat /opt/local/nginx/unit/unix/var/lib/unit/conf.json \
        | jq . \
        | grep unix;
            "unix:@abstract": {
                "unix:@abstract": {}
```

Reply:
Or try writing `"foo"` to that string, and see the config file after that.  You'll see that the config file is modified.

Reply:
I didn't see where we're editing it... I need to look at this thing in git...

Reply:
Yes, I see it now :(

Reply:
No problem!  It took me a long time to realize it myself  :-)